File: demo/common/src/main/java/com/serious/demo/TestComponent.java
package com.serious.demo;
/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

import com.serious.service.Component;
import com.serious.service.ComponentInterface;

/**
 * @author Andreas Ernst
 */
@ComponentInterface(
        name = "TestComponent",
        description = "du funny stuff",
        services = {
                TestService.class,
                TestRestService.class
        })
public interface TestComponent extends Component {
    String hello();
}

File: demo/common/src/main/java/com/serious/demo/TestComponent.kt
package com.serious.demo
/*
* @COPYRIGHT (C) 2023 Andreas Ernst
*
* All rights reserved
*/

import com.serious.service.Component
import com.serious.service.ComponentInterface

import jakarta.validation.constraints.Min
import jakarta.validation.constraints.Max
import jakarta.validation.constraints.NotNull

@ComponentInterface(
    name = "TestComponent",
    description = "do funny stuff",
    services = [TestService::class, TestRestService::class]
)
interface TestComponent : Component {
    @NotNull()
    fun hello(): String
}
